### PR TITLE
Smartagent Support

### PR DESCRIPTION
**Pull request by dsupure**
**Summary of changes:**
 :green_heart: I make the smartagent toolbox available for network debugging.  :green_heart:
So in the future, you can jump into all pso-db, pso-csi-controller-0, pso-csi-node pods to debug the network issue.
The command is like this:
```kubectl exec -n pso -it pso-db-0-0 -c pso-smartagent /bin/bash```

### CODE REVIEW

File: pure-pso/values.yaml [hunk 1]
```diff
@@ -95,14 +98,19 @@ images:
       pullPolicy: Always
     cockroachOperator:
       name: purestorage/cockroach-operator
-      tag: v1.0.3
+      tag: v1.0.4-smartagent-1
```
Comment (haibinxie) [inline pure-pso/values.yaml:101]: change back

File: pure-pso/values.yaml [hunk 2]
```diff
@@ -27,6 +27,9 @@ storagetopology:
   enable: false
   strictTopology: false
 
+# this option is to enable/disable the pso smartagent feature
```
Comment (haibinxie) [inline pure-pso/values.yaml:30]: this comment may cause some confusion as what smart agent is. we may need detailed documents to describe the container and its use cases.

**Comment by haibinxie** [comment]
I have a little concern about adding it in the official release:
1. it adds one more container to track in the release, which will make releases more complicated.
2. the db pod upgrade path does not work, it will cause confusion to customers.

Shall we consider just let user run "kubectl patch ..." to add the container as one of debugging steps, so we dont have to maintain it in official helm repo. and when we release the smart agent for debugging it can be more flexible.

File: pure-pso/values.yaml (hunk 1 above)
Comment (dsupure) [reply pure-pso/values.yaml:101]: Yes. I'll change back later. However, I leave it for now for the testing purpose.

**Comment by sdodsley** [comment]
@haibinxie I like that idea

**Comment by dsupure** [comment]
@haibinxie , @sdodsley: 
I know we need to maintain one more container, but this is an important one for debugging the network issue. 
I actually don't quite concern about the two points that @haibinxie mentioned. First, we only have a few containers to maintain today. Second, if users have a problem installing our tool, they won't need to do the `helm upgrade`.  

This is our preliminary step. In the future, we should have some scripts that can automate the network debugging and generate some report.

**Comment by haibinxie** [comment]
> @haibinxie , @sdodsley:
> I know we need to maintain one more container, but this is an important one for debugging the network issue.
> 
> I actually don't quite concern about the two points that @haibinxie mentioned. First, we only have a few containers to maintain today. Second, if users have a problem installing our tool, they won't need to do the `helm upgrade`.
> 
> This is our preliminary step. In the future, we should have some scripts that can automate the network debugging and generate some reports.
> 
> We can ask users to do the `kubectl patch`, but it's not user friendly. You need to patch 7 pods. Making it the default option in the `value.yaml` is much easier for them to install this toolbox.

Sounds fair, I am ok with it.

**Comment by Pure-AdamuKaapan** [comment]
RE the kubectl patch idea, you can do kubectl patch by label, so all PSO pods would get it. More containers to track and add is worth considering (more complex release process, more complex dark site install, more security audits, though it being optional does help) in my opinion. I also agree with the concern that it won't be updated in the PSO DB pods until they're regenerated, which limits debugging effectiveness for db pod networking issues.

If we still think those are all acceptable clauses, then I'm okay going forwards with it.

**Comment by haibinxie** [comment]
FYI, here is how to use patch, run `kubectl patch statefulset pso-db-1 --patch "$(cat ../tmp/patch.yaml)"` with below patch file.

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: curl-deployment
spec:
  selector:
    matchLabels:
      app: curlpod
  replicas: 1
  template:
    metadata:
      labels:
        app: curlpod
    spec:
      containers:
        - name: curlpod
          command:
            - sh
            - -c
            - while true; do sleep 1; done
          image: radial/busyboxplus:curl
```

**Comment by sdodsley** [comment]
@dsupure is this still required?

**Comment by dsupure** [comment]
No, I think @haibinxie has a better tool to troubleshoot the network issue. Let me close it.

**Pull request closed without merging**